csunplugged/resources/views/BarcodeChecksumPosterResourceGenerator.py

```diff
@@ -0,0 +1,36 @@
+"""Class for Barcode Checksum Poster resource generator."""
+
+from PIL import Image
+from utils.BaseResourceGenerator import BaseResourceGenerator
+
+
+class BarcodeChecksumPosterResourceGenerator(BaseResourceGenerator):
+    """Class for Grid resource generator."""
+
+    additional_valid_options = {
```

Comment:
why does this live here and not outside of the class/what is this magic?

Reply:
It's a class variable, available at BarcodeChecksumPosterResourceGenerator.additional_valid_options. It's the same for all instances of the class, hence why it is set here rather than __init__